deepspeed/module_inject/replace_module.py

```diff
@@ -327,13 +327,13 @@ def set_lm_head(module):
                                              checkpoint=checkpoint_file)
             pbar.update(1)
             gc.collect()
-        replaced_module = set_lm_head(replaced_module)
     else:
         replaced_module = replace_module(model=model,
                                          orig_class=orig_layer_impl,
                                          replace_fn=replace_fn,
                                          _replace_policy=config.injection_policy_tuple)
 
+    replaced_module = set_lm_head(replaced_module)
```

Comment:
We move "set_lm_head" outside if...else so that lm_head can be replaced without loading checkpoint.